### PR TITLE
Upgrade kafka-clients to 3.3.1

### PR DESCRIPTION
**Pull request by svroonland**
We seem to have missed some scala-steward upgrades.

### CODE REVIEW

File: zio-kafka/src/main/scala/zio/kafka/admin/AdminClient.scala [hunk 1]
```diff
@@ -606,6 +622,39 @@ object AdminClient {
...
+      fromKafkaFuture {
+        ZIO.attemptBlocking(
+          adminClient
+            .listConsumerGroupOffsets(groupSpecs.map { case (groupId, offsetsSpec) =>
+              (groupId, offsetsSpec.asJava)
+            }.asJava)
+            .partitionsToOffsetAndMetadata()
+        )
+      }
+        .map(_.asScala.filterNot { case (_, om) => om eq null }.toMap.bimap(TopicPartition(_), OffsetAndMetadata(_)))
```
Comment (guizmaii) [inline zio-kafka/src/main/scala/zio/kafka/admin/AdminClient.scala:640]:
```suggestion
        .map(_.asScala.filter { case (_, om) => om ne null }.toMap.bimap(TopicPartition(_), OffsetAndMetadata(_)))
```

File: zio-kafka/src/main/scala/zio/kafka/admin/AdminClient.scala [hunk 2]
```diff
@@ -606,6 +622,39 @@ object AdminClient {
...
+        ZIO.attemptBlocking(
+          adminClient
+            .listConsumerGroupOffsets(
+              groupSpecs.map { case (groupId, offsetsSpec) => (groupId, offsetsSpec.asJava) }.asJava,
+              options.asJava
+            )
+            .partitionsToOffsetAndMetadata()
+        )
+      }
+        .map(_.asScala.filterNot { case (_, om) => om eq null }.toMap.bimap(TopicPartition(_), OffsetAndMetadata(_)))
```
Comment (guizmaii) [inline zio-kafka/src/main/scala/zio/kafka/admin/AdminClient.scala:656]:
```suggestion
        .map(_.asScala.filter { case (_, om) => om ne null }.toMap.bimap(TopicPartition(_), OffsetAndMetadata(_)))
```

**Review by guizmaii — DISMISSED** [review]
See suggestions :)

**Comment by svroonland** [comment]
Thanks!

**Comment by svroonland** [comment]
@guizmaii If you could approve again?

**Review by guizmaii — DISMISSED** [review]

**Review by guizmaii — APPROVED** [review]

**Pull request merged**